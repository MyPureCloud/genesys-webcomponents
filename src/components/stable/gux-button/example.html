<h4>Enabled</h4>
<div>
  <gux-button title="default">Default</gux-button>
  <gux-button title="Primary" accent="primary">Primary</gux-button>
  <gux-button title="Secondary" accent="secondary">Secondary</gux-button>
  <gux-button title="Invalid accent" accent="invalid"
    >Invalid accent</gux-button
  >
  <gux-button title="Slotted element" accent="invalid">
    <span>Text in a span element</span>
  </gux-button>
  <gux-button title="Slotted element" accent="invalid">
    <span>Text in a span element</span>
    <span>More text in a span element</span>
  </gux-button>
</div>

<h4>Disabled</h4>
<div>
  <gux-button title="default" disabled>Default</gux-button>
  <gux-button title="Primary" accent="primary" disabled>Primary</gux-button>
  <gux-button title="Secondary" accent="secondary" disabled
    >Secondary</gux-button
  >
  <gux-button title="Invalid accent" accent="invalid" disabled
    >Invalid accent</gux-button
  >
  <gux-button title="Slotted element" accent="invalid" disabled>
    <span>Text in a span element</span>
  </gux-button>
  <gux-button title="Slotted element" accent="invalid" disabled>
    <span>Text in a span element</span>
    <span>More text in a span element</span>
  </gux-button>
</div>

<style>
  gux-button {
    margin: 10px;
  }
</style>
