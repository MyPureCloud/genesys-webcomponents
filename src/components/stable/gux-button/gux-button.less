@import (reference) '../../../style/color.less';

// Variables part
@primary-background-color: @gux-genesys-dark-blue;
@primary-border-color: #2455a3;
@primary-color: @gux-type-dark;
@primary-active-background-color: #275aad;

@primary-dark-background-color: @gux-genesys-dark-blue;
@primary-dark-border-color: @gux-grey-global;
@primary-dark-color: @gux-type-dark;
@primary-dark-hover-background-color: #3476f5;

@secondary-background-color: @gux-border;
@secondary-border-color: #d5dce0;
@secondary-color: @gux-type;
@secondary-active-background-color: #dce1e5;

@secondary-dark-background-color: #555d66;
@secondary-dark-border-color: @gux-grey-global;
@secondary-dark-color: @gux-type-dark;
@secondary-dark-hover-background-color: #76818f;

gux-button {
  display: inline-block;
  min-width: 32px;
  pointer-events: none;

  > button {
    width: 100%;
    padding: 0 16px;
    height: 32px;
    border-radius: 2px;
    cursor: pointer;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    font-size: 12px;
    font-weight: bold;
    pointer-events: auto;

    &[disabled] {
      opacity: 0.5;
      cursor: default;
    }

    &:focus {
      outline: none;
    }
  }
}

// Theming
.gux-button-dark-theme {
  > button {
    &.gux-secondary {
      background-color: @secondary-dark-background-color;
      border: 1px solid @secondary-dark-border-color;
      color: @secondary-dark-color;

      &:hover {
        background-color: @secondary-dark-hover-background-color;
      }

      &:active {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 1);
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          background-color: @secondary-dark-background-color;
        }

        &:active {
          box-shadow: none;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }

    &.gux-primary {
      background-color: @primary-dark-background-color;
      border: 1px solid @primary-dark-border-color;
      color: @primary-dark-color;

      &:hover {
        background-color: @primary-dark-hover-background-color;
      }

      &:active {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 1);
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          background-color: @primary-dark-background-color;
        }

        &:active {
          box-shadow: none;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }
  }
}

.gux-button-light-theme {
  > button {
    &.gux-secondary {
      background-color: @secondary-background-color;
      border: 1px solid @secondary-border-color;
      color: @secondary-color;

      &:hover {
        box-shadow: 0 0 1px 1px rgba(34, 37, 41, 0.24);
      }

      &:active {
        background-color: @secondary-active-background-color;
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          box-shadow: none;
        }

        &:active {
          background-color: @secondary-background-color;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }

    &.gux-primary {
      background-color: @primary-background-color;
      border: 1px solid @primary-border-color;
      color: @primary-color;

      &:hover {
        box-shadow: 0 0 1px 1px rgba(34, 37, 41, 0.48);
      }

      &:active {
        background-color: @primary-active-background-color;
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          box-shadow: none;
        }

        &:active {
          background-color: @primary-background-color;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }
  }
}

// Default Theme
gux-button {
  .gux-button-light-theme();
}

// Dark
.gux-dark-theme {
  gux-button {
    .gux-button-dark-theme();
  }
}

gux-button.gux-dark-theme {
  .gux-button-dark-theme();
}

// Light
.gux-light-theme {
  gux-button {
    .gux-button-light-theme();
  }
}

gux-button.gux-light-theme {
  .gux-button-light-theme();
}
