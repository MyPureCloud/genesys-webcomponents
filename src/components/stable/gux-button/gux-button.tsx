import {
  Component,
  ComponentInterface,
  Element,
  h,
  JSX,
  Prop
} from '@stencil/core';

import { logComponentUsage } from '../../../common/log-component-usage';

export type GuxButtonAccent = 'primary' | 'secondary';

@Component({
  styleUrl: 'gux-button.less',
  tag: 'gux-button'
})
export class GuxButton implements ComponentInterface {
  @Element()
  private root: HTMLElement;

  /**
   * The component title
   */
  @Prop()
  title: string;

  /**
   * Indicate if the button is disabled or not
   */
  @Prop()
  disabled = false;

  /**
   * The component accent (secondary or primary).
   */
  @Prop()
  accent: GuxButtonAccent = 'secondary';

  constructor() {
    logComponentUsage('gux-button');
  }

  componentWillLoad() {
    this.makeSlotContentDisableable();
  }

  render(): JSX.Element {
    return (
      <button
        title={this.title}
        disabled={this.disabled}
        class={this.accent === 'primary' ? 'gux-primary' : 'gux-secondary'}
      >
        <slot />
      </button>
    );
  }

  private makeSlotContentDisableable() {
    Array.from(this.root.children).forEach(slotElement => {
      slotElement.addEventListener('click', (event: MouseEvent): void => {
        if (this.disabled) {
          event.stopImmediatePropagation();
          event.stopPropagation();
          event.preventDefault();
        }
      });
    });
  }
}
