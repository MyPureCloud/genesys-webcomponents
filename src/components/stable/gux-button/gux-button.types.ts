export type GuxButtonAccent = 'primary' | 'secondary' | 'tertiary';

export type GuxButtonType = 'button' | 'submit' | 'reset';
