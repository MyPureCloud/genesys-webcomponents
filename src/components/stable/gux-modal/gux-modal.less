@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/color.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/mixins.less';

// Style
gux-modal {
  .gux-modal {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1;
    display: flex;
    align-items: center;
    justify-content: center;
    color: @gux-black-50;
    background: fade(@gux-black-30, 80%);

    .gux-modal-container {
      .shadow-20();

      position: relative;
      box-sizing: border-box;
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      padding: 40px;
      background: @gux-grey-100;
      border: 1px solid @gux-black-100;

      &.gux-small {
        width: 416px;
        max-height: e(%('min(368px, 100vh - 2 * %s)', @spacing-large));
      }

      &.gux-medium {
        width: 672px;
        max-height: e(%('min(640, 100vh - 2 * %s)', @spacing-large));
      }

      &.gux-large {
        width: 940px;
        max-height: e(%('min(640, 100vh - 2 * %s)', @spacing-large));
      }

      &.gux-dynamic {
        max-width: calc(100vw - 2 * @spacing-large);
        max-height: calc(100vh - 2 * @spacing-large);

        .gux-modal-content {
          max-height: none;
        }
      }

      .gux-dismiss-button {
        position: absolute;
        top: 15px;
        right: 15px;
        padding: 0;
        font-size: 11px;
        color: @gux-black-90;
        cursor: pointer;
        background: transparent;
        border: none;

        gux-icon {
          width: 12px;
          height: 12px;
        }
      }

      .gux-modal-header {
        padding-bottom: 24px;
        margin: 0;
        .h1();
      }

      .gux-modal-content {
        max-height: 432px;
        margin: 0 8px 24px 8px;
        overflow-y: auto;

        &.gux-no-buttons {
          margin-bottom: 0;
        }
      }

      .gux-button-footer {
        display: flex;
        justify-content: space-between;

        .gux-left-align-buttons {
          gux-button {
            padding-right: 5px;
          }
        }

        .gux-right-align-buttons {
          gux-button {
            padding-left: 5px;
          }
        }
      }
    }

    .gux-modal-container {
      @media (max-width: 416px) {
        &.gux-small,
        &.gux-medium,
        &.gux-large {
          width: 100%;
          height: 100%;
        }
      }
    }
  }
}

// Theming

// Dark
.gux-modal-dark-theme {
  color: @gux-type-dark;

  .gux-modal-container {
    background: @gux-black-50;
  }
}

.gux-dark-theme {
  gux-modal {
    .gux-modal-dark-theme();
  }
}

gux-modal.gux-dark-theme {
  .gux-modal-dark-theme();
}
