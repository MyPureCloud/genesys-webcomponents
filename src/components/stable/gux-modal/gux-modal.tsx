import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop
} from '@stencil/core';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';

import modalComponentResources from './i18n/en.json';

export type GuxModalSize = 'small' | 'medium' | 'large';

/**
 * @slot content - Required slot for the modal content
 * @slot left-align-buttons - Optional slot to set gux-buttons aligned to the left of the modal
 * @slot right-align-buttons - Optional slot to set gux-buttons aligned to the left of the modal
 * @slot title - Optional slot to set the modal title
 */
@Component({
  styleUrl: 'gux-modal.less',
  tag: 'gux-modal'
})
export class GuxModal {
  /**
   * Indicates the size of the modal (small, medium or large)
   */
  @Prop()
  size: GuxModalSize = 'small';

  /**
   * Fired when a user dismisses the modal
   */
  @Event()
  guxdismiss: EventEmitter<void>;
  private getI18nValue: GetI18nValue;

  @Element()
  private element: HTMLElement;

  async componentWillLoad(): Promise<void> {
    this.getI18nValue = await buildI18nForComponent(
      this.element,
      modalComponentResources
    );
  }

  render(): JSX.Element {
    const hasModalTitleSlot = this.hasModalTitleSlot();
    const hasFooterButtons = this.hasFooterButtons();

    return (
      <div class="modal">
        <div class={`modal-container ${this.size}`}>
          <button
            class="dismiss-button"
            title={this.getI18nValue('dismiss')}
            onClick={this.onDismissClickHandler.bind(this)}
          >
            <gux-icon
              screenreader-text={this.getI18nValue('dismiss')}
              icon-name="ic-close"
            ></gux-icon>
          </button>

          {hasModalTitleSlot && (
            <h1 class="modal-header">
              <slot name="title" />
            </h1>
          )}

          <div
            class={{ 'modal-content': true, 'no-buttons': !hasFooterButtons }}
          >
            <p>
              <slot name="content" />
            </p>
          </div>

          {hasFooterButtons && (
            <div class="button-footer">
              <div class="left-align-buttons">
                <slot name="left-align-buttons" />
              </div>

              <div class="right-align-buttons">
                <slot name="right-align-buttons" />
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }

  private hasModalTitleSlot(): boolean {
    return Boolean(this.element.querySelector('[slot="title"]'));
  }

  private hasFooterButtons(): boolean {
    return (
      Boolean(this.element.querySelector('[slot="left-align-buttons"]')) ||
      Boolean(this.element.querySelector('[slot="right-align-buttons"]'))
    );
  }

  private onDismissClickHandler(event: MouseEvent): void {
    event.stopPropagation();

    const dismissEvent = this.guxdismiss.emit();
    if (!dismissEvent.defaultPrevented) {
      this.element.remove();
    }
  }
}
