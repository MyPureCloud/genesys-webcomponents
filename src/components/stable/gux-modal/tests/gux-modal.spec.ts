import { newSpecPage } from '@stencil/core/testing';
import { GuxButton } from '../../gux-button/gux-button';
import { GuxModal } from '../gux-modal';

const components = [GuxButton, GuxModal];
const language = 'en';

describe('gux-modal', () => {
  describe('#render', () => {
    [
      {
        description: 'should render small modal',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render medium modal',
        html: `
          <gux-modal size="medium">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render large modal',
        html: `
          <gux-modal size="large">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without a title',
        html: `
          <gux-modal size="large">
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just left align buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just right align buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render small modal by default',
        html: `
          <gux-modal>
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxModal);

        expect(page.root).toMatchSnapshot();

        page.rootInstance.hidden = true;
        await page.waitForChanges();

        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('dismiss', () => {
    it('click dismiss button', async () => {
      const html = `
        <gux-modal size="small">
          <div slot="title">Modal Title</div>
          <div slot="content">This contains the modal content.</div>
          <div slot="left-align-buttons">
              <gux-button title="Cancel">Cancel</gux-button>
          </div>
          <div slot="right-align-buttons">
            <gux-button title='Button' accent='primary'>Accept</gux-button>
          </div>
        </gux-modal>
      `;
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const dismissButton = page.root.querySelector(
        '.gux-dismiss-button'
      ) as HTMLButtonElement;
      const guxdismissSpy = jest.fn();
      const clickSpy = jest.fn();
      const elementRemoveSpy = jest.spyOn(element, 'remove');

      page.win.addEventListener('guxdismiss', guxdismissSpy);
      page.win.addEventListener('click', clickSpy);

      dismissButton.click();
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveBeenCalled();
      expect(clickSpy).not.toHaveBeenCalled();
      expect(elementRemoveSpy).toBeCalledWith();
    });

    it('click dismiss button and prevent default', async () => {
      const html = `
        <gux-modal size="small">
          <div slot="title">Modal Title</div>
          <div slot="content">This contains the modal content.</div>
          <div slot="left-align-buttons">
              <gux-button title="Cancel">Cancel</gux-button>
          </div>
          <div slot="right-align-buttons">
            <gux-button title='Button' accent='primary'>Accept</gux-button>
          </div>
        </gux-modal>
      `;
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const dismissButton = page.root.querySelector(
        '.gux-dismiss-button'
      ) as HTMLButtonElement;
      const elementRemoveSpy = jest.spyOn(element, 'remove');

      page.win.addEventListener('guxdismiss', (event: Event) => {
        event.preventDefault();
      });

      dismissButton.click();
      await page.waitForChanges();

      expect(elementRemoveSpy).not.toBeCalled();
    });
  });
});
