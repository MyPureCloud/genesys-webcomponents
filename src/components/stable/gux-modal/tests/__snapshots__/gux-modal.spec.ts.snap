// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-modal #render should render large modal 1`] = `
<gux-modal size="large">
  <!---->
  <div class="modal">
    <div class="large modal-container">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render large modal 2`] = `
<gux-modal size="large">
  <!---->
  <div class="modal">
    <div class="large modal-container">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render medium modal 1`] = `
<gux-modal size="medium">
  <!---->
  <div class="modal">
    <div class="medium modal-container">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render medium modal 2`] = `
<gux-modal size="medium">
  <!---->
  <div class="modal">
    <div class="medium modal-container">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just left align buttons 1`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons"></div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just left align buttons 2`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons"></div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just right align buttons 1`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons"></div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just right align buttons 2`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons"></div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without a title 1`] = `
<gux-modal size="large">
  <!---->
  <div class="modal">
    <div class="large modal-container">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without a title 2`] = `
<gux-modal size="large">
  <!---->
  <div class="modal">
    <div class="large modal-container">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without buttons 1`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content no-buttons">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without buttons 2`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content no-buttons">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal 1`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal 2`] = `
<gux-modal size="small">
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal by default 1`] = `
<gux-modal>
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal by default 2`] = `
<gux-modal>
  <!---->
  <div class="modal">
    <div class="modal-container small">
      <button class="dismiss-button" title="Dismiss">
        <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
      </button>
      <h1 class="modal-header">
        <div slot="title">
          Modal Title
        </div>
      </h1>
      <div class="modal-content">
        <p>
          <div slot="content">
            This contains the modal content.
          </div>
        </p>
      </div>
      <div class="button-footer">
        <div class="left-align-buttons">
          <div slot="left-align-buttons">
            <gux-button title="Cancel">
              <!---->
              <button class="gux-secondary" title="Cancel">
                Cancel
              </button>
            </gux-button>
          </div>
        </div>
        <div class="right-align-buttons">
          <div slot="right-align-buttons">
            <gux-button accent="primary" title="Button">
              <!---->
              <button class="gux-primary" title="Button">
                Accept
              </button>
            </gux-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</gux-modal>
`;
